src/Xamarin.Android.Build.Tasks/Tasks/ConvertCustomView.cs

```diff
@@ -0,0 +1,152 @@
+		[Required]
+		public string AcwMapFile { get; set; }
+
+		public string ResourceNameCaseMap { get; set; }
+
+		public ITaskItem [] ResourceDirectories { get; set; }
+
+		public override bool Execute ()
+		{
+			var resource_name_case_map = MonoAndroidHelper.LoadResourceCaseMap (ResourceNameCaseMap);
+			var acw_map = MonoAndroidHelper.LoadAcwMapFile (AcwMapFile);
+			var customViewMap = MonoAndroidHelper.LoadCustomViewMapFile (BuildEngine4, CustomViewMapFile);
+			var processed = new HashSet<string> ();
+
+			foreach (var kvp in acw_map) {
+				var key = kvp.Key;
+				var value = kvp.Value;
+				if (key == value)
+					continue;
+				if (customViewMap.TryGetValue (key, out HashSet<string> resourceFiles)) {
+					foreach (var file in resourceFiles) {
+						if (processed.Contains (file))
+							continue;
+						if (!File.Exists (file))
+							continue;
+						var document = XDocument.Load (file);
+						var e = document.Root;
+						bool update = false;
+						foreach (var elem in AndroidResource.GetElements (e).Prepend (e)) {
+							update |= TryFixCustomView (elem, acw_map, (t, m) => {
+								string targetfile = file;
+								ITaskItem resdir = ResourceDirectories?.FirstOrDefault (x => file.StartsWith (x.ItemSpec)) ?? null;
+								if (resdir != null && targetfile.StartsWith (resdir.ItemSpec, StringComparison.InvariantCultureIgnoreCase)) {
+									targetfile = file.Substring (resdir.ItemSpec.Length).TrimStart (Path.DirectorySeparatorChar);
+									if (resource_name_case_map.TryGetValue (targetfile, out string temp))
+										targetfile = temp;
+									targetfile = Path.Combine ("Resources", targetfile);
+								}
+								switch (t) {
+								case TraceLevel.Error:
+									Log.LogCodedError ("XA1002", file: targetfile, lineNumber: 0, message: m);
+									break;
+								case TraceLevel.Warning:
+									Log.LogCodedWarning ("XA1001", file: targetfile, lineNumber: 0, message: m);
+									break;
+								default:
+									Log.LogDebugMessage (m);
+									break;
+								}
+							});
+						}
+						foreach (XAttribute a in AndroidResource.GetAttributes (e)) {
+							update |= TryFixCustomClassAttribute (a, acw_map);
+							update |= TryFixFragment (a, acw_map);
+						}
+						if (update) {
+							document.Save (file);
+						}
+						processed.Add (file);
+					}
+				}
+			}
+
+			return !Log.HasLoggedErrors;
+		}
+
+		static readonly XNamespace res_auto = "http://schemas.android.com/apk/res-auto";
+		static readonly XNamespace android = "http://schemas.android.com/apk/res/android";
+
+		bool TryFixCustomClassAttribute (XAttribute attr, Dictionary<string, string> acwMap)
+		{
+			/* Some attributes reference a Java class name.
+			 * try to convert those like for TryFixCustomView
+			 */
+			if (attr.Name != (res_auto + "layout_behavior")                              // For custom CoordinatorLayout behavior
+			    && (attr.Parent.Name != "transition" || attr.Name.LocalName != "class")) // For custom transitions
```

Comment:
Code convention: if the condition for an `if`/etc. spans multiple lines, it should be indented *one further tab stop* than the "body" of the `if` block.  (This is the same as XML attributes vs. XML children.)

Thus:

```csharp
if (attr.Name != (res_auto + "layout_behavior")                                    // For custom CoordinatorLayout behavior
        && (attr.Parent.Name != "transition" || attr.Name.LocalName != "class")) { // For custom transitions
    return false;
}
```

Additionally, we more frequently put the `&&`/`||` at the end of the preceding line (88 times in `src`) vs. 7 at the beginning of the line:

```bash
$ git grep -E '&&$' src | wc -l
      88
$ git grep -E '^\s+&&' src | wc -l
       7
```